### ISSUE TITLE
Get the name of a session

### ISSUE THREAD

**Issue by mikeholler** [theacodes/nox#415]
**How would this feature be useful?**
I would like to use the name of a session to create named reports folders, but I could see it being used a number of ways.

**Describe the solution you'd like**

```
import nox

@nox.session
def my_session(session: Session):
    reports_dir = Path("build", session.name)
    reports_dir.mkdirs()  # create build/my_session directory
    ...


@nox.session
@nox.parametrized("example", [
    nox.param("123"),
    nox.param("456"),
])
```

**Describe alternatives you've considered**

I was able to find a session name by writing this function and calling it in my session functions.

```
def session_name(session: Session) -> str:
    penv: ProcessEnv = session.virtualenv
    if isinstance(penv, VirtualEnv):
        return Path(penv.location).name
    else:
        raise NotImplementedError(
            f"Name support for type not yet implemented: {type(penv)}"
        )
```

**Comment by cjolowicz** [comment]
Hi @mikeholler , thanks for the feature request!

`Session.name` was recently implemented (#386). This has been merged into the main branch, but not yet released.

Could you try with the main branch of Nox and see if this works for you?

**Comment by cjolowicz** [comment]
This was resolved in #386. Closing for now, happy to reopen if there's anything left for us to do here.
